### PR TITLE
Add a button for disabling all mods to world config.

### PR DESCRIPTION
**Pull request by red-001**

### CODE REVIEW

**Comment by ObaniGemini** [comment]
Good idea, but low priority :)

File: builtin/mainmenu/dlg_config_world.lua [hunk 1]
```diff
@@ -80,14 +83,18 @@ local function get_formspec(data)
 			end
 		end
 	end
-
+	if enabled_all then
```
Comment (kilbith) [inline builtin/mainmenu/dlg_config_world.lua]: Whitespace at end and you need to put one more indentation level on the following code.

File: builtin/mainmenu/dlg_config_world.lua [hunk 2]
```diff
@@ -80,14 +83,18 @@ local function get_formspec(data)
...
-
+	else
+	retval = retval ..
+		"button[8.5,-0.125;2.5,0.5;btn_enable_all_mods;" .. fgettext("Enable all") .. "]" ..
+		"textlist[5.5,0.5;5.5,5.75;world_config_modlist;"
+	end
 	retval = retval .. modmgr.render_modlist(data.list)
 	retval = retval .. ";" .. data.selected_mod .."]"
-
+
```
Comment (kilbith) [inline builtin/mainmenu/dlg_config_world.lua]: Grrr...

File: builtin/mainmenu/dlg_config_world.lua [hunk 3]
```diff
@@ -229,7 +236,7 @@ local function handle_buttons(this, fields)
 		return true
 	end
 
-	if fields["btn_all_mods"] then
+	if fields["btn_enable_all_mods"] then
```
Comment (kilbith) [inline builtin/mainmenu/dlg_config_world.lua]:
``` Lua
if fields.btn_enable_all_mods then
```

File: builtin/mainmenu/dlg_config_world.lua [hunk 4]
```diff
@@ -238,6 +245,20 @@ local function handle_buttons(this, fields)
...
 			end
 		end
+		enabled_all = true
+		return true
+	end
+	
+	if fields["btn_disable_all_mods"] then
+		local list = this.data.list:get_raw_list()
+
+		for i=1,#list,1 do
```
Comment (kilbith) [inline builtin/mainmenu/dlg_config_world.lua]: Don't need the `,1`. Also spaces missing.

File: builtin/mainmenu/dlg_config_world.lua [hunk 5]
```diff
@@ -238,6 +245,20 @@ local function handle_buttons(this, fields)
...
+			if list[i].typ ~= "game_mod" and
+				not list[i].is_modpack then
```
Comment (kilbith) [inline builtin/mainmenu/dlg_config_world.lua]:
``` Lua
if list[i].typ ~= "game_mod" and not
      list[i].is_modpack then
```

Comment (red-001) [inline builtin/mainmenu/dlg_config_world.lua]: I don't think this line needs to be broken in the first place it is under 80 characters.

Comment (kilbith) [inline builtin/mainmenu/dlg_config_world.lua]: It's neither right or wrong, that's just the guideline to break after the `not`.

File: builtin/mainmenu/dlg_config_world.lua [hunk 6]
```diff
@@ -80,11 +83,15 @@ local function get_formspec(data)
...
-
+	if enabled_all then 
+		retval = retval ..
+			"button[8.5,-0.125;2.5,0.5;btn_disable_all_mods;" .. fgettext("Disable all") .. "]" ..
+			"textlist[5.5,0.5;5.5,5.75;world_config_modlist;"
+	else
+		retval = retval ..
+			"button[8.5,-0.125;2.5,0.5;btn_enable_all_mods;" .. fgettext("Enable all") .. "]" ..
+			"textlist[5.5,0.5;5.5,5.75;world_config_modlist;"
+		end
```
Comment (kilbith) [inline builtin/mainmenu/dlg_config_world.lua]: One less indent.

File: builtin/mainmenu/dlg_config_world.lua [hunk 7]
```diff
@@ -229,15 +236,27 @@ local function handle_buttons(this, fields)
...
 	end
 
-	if fields["btn_all_mods"] then
+	if fields.btn_enable_all_mods then
 		local list = this.data.list:get_raw_list()
 
-		for i=1,#list,1 do
-			if list[i].typ ~= "game_mod" and
-				not list[i].is_modpack then
+		for i=1,#list do
```
Comment (kilbith) [inline builtin/mainmenu/dlg_config_world.lua]: Spaces.

File: builtin/mainmenu/dlg_config_world.lua [hunk 8]
```diff
@@ -229,15 +236,27 @@ local function handle_buttons(this, fields)
...
 			end
 		end
+		enabled_all = true
+		return true
+	end
+	
+	if fields.btn_disable_all_mods then
+		local list = this.data.list:get_raw_list()
+
+		for i=1,#list do
```
Comment (kilbith) [inline builtin/mainmenu/dlg_config_world.lua]: Ditto.

File: builtin/mainmenu/dlg_config_world.lua [hunk 9]
```diff
@@ -229,15 +236,27 @@ local function handle_buttons(this, fields)
...
 	end
 
-	if fields["btn_all_mods"] then
+	if fields.btn_enable_all_mods then
 		local list = this.data.list:get_raw_list()
 
-		for i=1,#list,1 do
-			if list[i].typ ~= "game_mod" and
-				not list[i].is_modpack then
+		for i=1, #list do
```
Comment (kilbith) [inline builtin/mainmenu/dlg_config_world.lua]: Spaces around `=`.

**Comment by paramat** [comment]
:-1: Why? All mods start disabled. I don't feel there's enough need.

**Comment by red-001** [comment]
So you don't have to disable every mod individually if you use the 'enable all' button?

**Comment by C1ffisme** [comment]
@paramat Some people accidentally hit the button, or some people might just click it and think "The more mods, the better!" without thinking about conflicts.

**Comment by paramat** [comment]
Ah sorry i forgot we had 'enable all'.

**Comment by red-001** [comment]
Could this be merged or closed?

**Comment by paramat** [comment]
I recommend keeping it open, it's a reasonable request.

**Pull request merged**